### PR TITLE
Landing Review Diff Mode (behind feature flag)

### PR DESCRIPTION
**Pull request by acunniffe**
Distribution aware interpreters started as a background project during C-cycle, we're trying to land it now during D-cycle. 

The sin of 'long-lived feature branches' was committed here. My penance is this PR that lands `/review` behind a feature flag. Esp with the new WASM diff hitting develop, I want to be working off the latest all the time. We'll see the Review diff page in development now, but it will not show up in @LouManglass beta or @latest channels just yet. The legacy diff page is there, just manually go to `/diffs` manually for the same experience you're used to. 

There's a few steps of hardening left to do before we want to put this in @beta 
- start using the WASM diff engine, then finish the work of hardening the interpreters with tests/snapshots
- setup sentry to catch diff/interpreter issues found in the wild
^ presuming @JaapRood 's wasm work continues, I can probably finish this before week's end.  

Nice to have's that can come in post-shipping to @beta:
- Removing existing diff page and deleting 60% of the codebase....
- Shipping Rust Body Learners for initial bodies and to power the diff UI.

### CODE REVIEW

File: workspaces/ui/config/webpack.config.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const ModuleNotFoundPlugin = require('react-dev-utils/ModuleNotFoundPlugin');
 const ForkTsCheckerWebpackPlugin = require('react-dev-utils/ForkTsCheckerWebpackPlugin');
 const typescriptFormatter = require('react-dev-utils/typescriptFormatter');
 const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
-const DiffEngineWebpackPlugin = require('@useoptic/diff-engine-wasm/webpack.plugin');
+const ThreadsPlugin = require('threads-plugin');
```
Comment (JaapRood) [inline workspaces/ui/config/webpack.config.js:29]: This has not been merged properly

File: workspaces/ui/config/webpack.config.js [hunk 2]
```diff
@@ -466,12 +466,7 @@ module.exports = function (webpackEnv) {
               // its runtime that would otherwise be processed through "file" loader.
               // Also exclude `html` and `json` extensions so they get processed
               // by webpacks internal loaders.
-              exclude: [
-                /\.(js|mjs|jsx|ts|tsx)$/,
-                /\.html$/,
-                /\.json$/,
-                /\.wasm$/,
-              ],
+              exclude: [/\.(js|mjs|jsx|ts|tsx)$/, /\.html$/, /\.json$/],
```
Comment (JaapRood) [inline workspaces/ui/config/webpack.config.js:469]: More not merged properly.

File: workspaces/ui/config/webpack.config.js [hunk 3]
```diff
@@ -509,8 +504,7 @@ module.exports = function (webpackEnv) {
             : undefined
         )
       ),
-      // Triggers and watches diff-engine-rust to wasm + browser glue compilation
-      DiffEngineWebpackPlugin(),
+      new ThreadsPlugin(),
       // Inlines the webpack runtime script. This script is too small to warrant
```
Comment (JaapRood) [inline workspaces/ui/config/webpack.config.js:508]: Also not merged properly.

**Review by JaapRood — CHANGES_REQUESTED** [review]
Was helping Lou with some debugging and checked this out for changes in the WASM, found out that the web pack stuff wasn't merged correctly.

**Comment by acunniffe** [comment]
@JaapRood  -- I fixed the merge issues that slipped through the cracks and can confirm the code here works nicely with the WASM engine now. ExampleDiffService changed a bit as part of this work, but nothing major has shifted underneath you.

**Pull request merged**